GomokuClient/packages/gomoku-story/src/components/TimeControls/TimeControls.tsx

```diff
@@ -12,45 +11,60 @@ export interface GameType {
 
 export interface TimeControlsProps {
   gameTypes: GameType[];
-  onCreateGame: (
-    boardSize: number,
-    timeControl?: SwaggerTypes.TimeControlDto,
-  ) => void;
+  onGameTypeSelected: (gameType: GameType) => void;
+  onGameTypeUnselected: () => void;
   isLoading: boolean;
 }
 
 export const TimeControls = ({
   gameTypes,
-  onCreateGame,
-  isLoading,
-}: TimeControlsProps) => (
-  <div className="grid grid-cols-2 gap-4 sm:grid-cols-3 sm:gap-6">
-    {gameTypes.map((game, index) => (
-      <GameCreatorButton
-        key={index}
-        timeControl={game.timeControl}
-        onCreateGame={onCreateGame}
-        isLoading={isLoading}
-      >
+  onGameTypeSelected,
+  onGameTypeUnselected,
+}: TimeControlsProps) => {
+  const [selectedGameType, setSelectedGameType] = useState<GameType>();
+
+  useEffect(() => {
+    return () => onGameTypeUnselected();
+  }, []);
+
+  return (
+    <div className="grid grid-cols-2 gap-4 sm:grid-cols-3 sm:gap-6">
+      {gameTypes.map((game, index) => (
         <Card
-          className="cursor-pointer border-[#2b2b2b] bg-[#2b2b2b] transition-colors
-            hover:bg-[#3e3e3e]"
+          key={index}
```

Comment:
1 and 2 not this case. List is static. Can't disagree with point 3, but what to use as key, if we don't have any id or something?

Reply:
It’s generally better to avoid using the index as a key. While there may be situations where you’re short on time and leave it as static, this approach can introduce unnecessary complexity. Over time, as the list evolves and becomes dynamic, it’s easy to overlook this and face issues. It’s best to follow standard patterns and avoid relying on highly specific cases where using the index as a key might be acceptable.

In this specific case, we can avoid using the index by combining multiple properties from the gameTypes object, such as:

```ts
{
  timeLabel: "1+0",
  type: t`Bullet`,
  boardSize: 13,
  timeControl: {
    initialTimeInSeconds: 60,
    incrementPerMove: 0,
  },
},
```

For example, you can create a unique key by combining properties like timeLabel, type, boardSize, and incrementPerMove. A combination like game.type + game.boardSize would work well. This approach ensures that the key remains stable and meaningful, even if the list changes over time

Reply:
Thanks for clarification. Removed indexed key